anndata/tests/test_io_warnings.py

```diff
@@ -43,6 +45,13 @@ def test_old_format_warning_not_thrown(tmp_path):
 
     with warnings.catch_warnings(record=True) as record:
         warnings.simplefilter("always", ad.OldFormatWarning)
+        if Version(h5py.__version__) < Version("3.2"):
```

Comment:
```suggestion
        if Version(version("h5py")) < Version("3.2"):
```

let’s use the standard API (`importlib.metadata:version`) and not a convention

Reply:
I don't think we should hit the disk for a version check of an imported package, as a rule.

I would agree that this could be better if this were asking the version of an dynamically selected package, but we are asking for the version of a package we know.

Reply:
As a rule, I ask for numbers before I consider “let’s use the hack instead of the dedicated API”. The numbers say: “Both add <1ms to function execution time, so don’t worry”

```console
$ python -m timeit -s "from importlib.metadata import version; version('pandas')" "version('pandas')"
1000 loops, best of 5: 225 usec per loop
$ python -m timeit -s "import pandas as pd" "pd.__version__"
10000000 loops, best of 5: 22.6 nsec per loop
```

Reply:
Did you try on the HPC? File access is randomly incredibly slow. E.g. seconds.

Reply:
Also, did the correctness issues on version numbers reported by importlib ever get fixed?

Reply:
> File access is randomly incredibly slow. E.g. seconds.

So Python importing is extremely slow too, which we do at run time in some cases. So this doesn‘t noticably exacerbate the problem.

> Also, did the correctness issues on version numbers reported by importlib ever get fixed?

Is there a bug I’m not aware of?

Reply:
> Is there a bug I’m not aware of?

There are issues that we've discussed before: https://github.com/scverse/anndata/pull/1001#issuecomment-1593163740

As an example in an environment right now:

```sh
$ python3 -c "import anndata; from importlib.metadata import version; print(anndata.__version__, version('anndata'))"
0.10.4.dev0+g2ee11c4.d20240118 0.10.1
```

See also the thread on the python discourse I linked above.

> We can fix this specific symptom of the underlying problem by defining PANDAS_VERSION

As discussed in person, pandas provides this already as `pd.__version__`. Though this variable does not suffer from the correctness problems of `importlib.metadata.version` with editable or modified installs.

Reply:
> and we should switch our stuff away from accessing .__version__ anywhere.

How about you open an issue for this and we use the convention we already use multiple places in this library (`.__version__`) here?

Reply:
Sure!